autoware_cmake/cmake/autoware_package.cmake

```diff
@@ -0,0 +1,45 @@
+# Copyright 2022 The Autoware Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+macro(autoware_package)
```

Comment:
I find the name of this macro a bit confusing, because later on in a package we have to call `ament_auto_package()` or `ament_package()`.  I think a better name would be `autoware_configure_options()`, `autoware_set_definitions()`, etc., but something that doesn't look like `ament_package()`

Reply:
I referred to `nav2` about the naming.

https://github.com/ros-planning/navigation2/blob/da602f8a19fa6aaa7078e56ff0c3c83517feb533/nav2_common/cmake/nav2_package.cmake#L20
https://github.com/ros-planning/navigation2/blob/da602f8a19fa6aaa7078e56ff0c3c83517feb533/nav2_controller/CMakeLists.txt#L18

I feel `autoware_configure_options` is good, but I'll investigate similar packages a bit more.

Reply:
MoveIt2 also uses `_package`.

https://github.com/ros-planning/moveit2/blob/main/moveit_common/cmake/moveit_package.cmake
https://github.com/ros-planning/moveit2/blob/002dcbf82d0b95a28ac17048c16b8f00679d0e66/moveit_planners/pilz_industrial_motion_planner/CMakeLists.txt#L6

Reply:
They were added here.
https://github.com/ros-planning/navigation2/commit/3f65576a98330de0cb832d2c344474b2b604142d
https://github.com/ros-planning/moveit2/commit/9a479dfae3a1a4ae2df6bfd094707c09eeaaef9b

Reply:
Not a huge fan of the name, but if it's a convention among other packages, in that case I prefer following it instead of coming up with something new.

Reply:
I understand, thank you. Regarding `nav2` and `MoveIt`, I think it's not a strong convention.
I guess, if we add more `.cmake` files in the future, we have to consider better namings. Let's re-consider it at that time if necessary. :+1:

Reply:
@kenji-miyake function is less invasive than macro, it's preferred to use function because that way the code in the function won't interact with the calling code. If there's no need to change the surrounding code, I'd suggest to change autoware_package to a function